cmd/kubeadm/app/preflight/checks.go

```diff
@@ -1021,6 +1013,12 @@ func InitNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.InitConfiguratio
 // The boolean flag 'downloadCerts' controls whether we should skip checks on certificates because we are downloading them.
 // If the flag is set to true we should skip checks already executed by RunJoinNodeChecks.
 func RunInitNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.InitConfiguration, ignorePreflightErrors sets.Set[string], isSecondaryControlPlane bool, downloadCerts bool) error {
+	if !isSecondaryControlPlane {
+		// First, check if we're root separately from the other preflight checks and fail fast.
+		if err := RunRootCheckOnly(ignorePreflightErrors); err != nil {
+			return err
+		}
+	}
```

Comment:
instead of calling the root check inside the runinit|joinnodechecks we should call RunRootCheckOnly in these places:

```
grep -E 'RunJoinNodeChecks|RunInitNodeChecks' * -rnI | grep -v chec
ks.go
app/cmd/phases/join/preflight.go:94:    if err := preflight.RunJoinNodeChecks(utilsexec.New(), j.Cfg(), j.IgnorePreflightErrors()); err != nil {
app/cmd/phases/join/preflight.go:123:           if err := preflight.RunInitNodeChecks(utilsexec.New(), initCfg, j.IgnorePreflightErrors(), true, hasCertificateKey); err != nil {
app/cmd/phases/init/preflight.go:65:    if err := preflight.RunInitNodeChecks(utilsexec.New(), data.Cfg(), data.IgnorePreflightErrors(), false, false); err != nil {
```

before the run init/join checks are called.

Reply:
the flag isSecondaryControlPlane should be removed from RunInitNodeChecks

actually 
app/cmd/phases/join/preflight.go:123 - should not call RunRootCheck
app/cmd/phases/init/preflight.go:65 - should call RunRootCheck

this is confirmed by the flag values which are passed ( isSecondaryControlPlane)

Reply:
Alright, I called in the sub-functions.

Reply:
the way you did it is also OK, but i think we should be more explicit about the runned root checks, similarly to how kubeadm upgrade and reset does it.

Reply:
Updated. It seems it is not required for 
```
app/cmd/phases/join/preflight.go:123:           if err := preflight.RunInitNodeChecks(utilsexec.New(), initCfg, j.IgnorePreflightErrors(), true, hasCertificateKey); err != nil {
```
as `isSecondaryControlPlane` is `true` here.